service/history/mutableStateBuilder.go

```diff
@@ -230,14 +229,14 @@ func (e *mutableStateBuilder) FlushBufferedEvents() error {
 	return nil
 }
 
-func (e *mutableStateBuilder) ApplyReplicationStateUpdates(failoverVersion int64) {
+func (e *mutableStateBuilder) ApplyReplicationStateUpdates(failoverVersion, lastEventID int64) {
 	e.replicationState.CurrentVersion = failoverVersion
```

Comment:
where is the logic that you prevent failover from version that is lower than current version?

Reply:
Could you help answer this?

Reply:
This change just implements the happy path.  I will be implementing that part of the replication protocol in a separate PR.